docs/migration_guides/algorithms_migration.rst

```diff
@@ -0,0 +1,790 @@
+    * New import path (Primitives): ``from qiskit.algorithms.minimum_eigensolvers import VQE, SamplingVQE, QAOA, NumPyMinimumEigensolver``
+
+VQE
+~~~
+
+The legacy :class:`qiskit.algorithms.minimum_eigen_solvers.VQE` class has now been split according to the use-case:
+
+- For general-purpose hamiltonians, you can use the Estimator-based :class:`qiskit.algorithms.minimum_eigensolvers.VQE`
+  class.
+- If you have a diagonal hamiltonian, and would like the algorithm to return a sampling of the state, you can use
+  the new Sampler-based :class:`qiskit.algorithms.minimum_eigensolvers.SamplingVQE` algorithm. This could formerly
+  be realized using the legacy :class:`~qiskit.algorithms.minimum_eigen_solvers.VQE` with
+  :class:`~qiskit.opflow.expectations.CVaRExpectation`.
+
+.. note::
+
+    The new :class:`~qiskit.algorithms.minimum_eigensolvers.VQEResult` class does not include the state anymore, as
+    this output was only useful in the case of diagonal operators. However, if it is available as part of the new
+    :class:`~qiskit.algorithms.minimum_eigensolvers.SamplingVQE` :class:`~qiskit.algorithms.minimum_eigensolvers.SamplingVQEResult`.
+
+.. raw:: html
+
+    <details>
+    <summary><a><font size="+1">VQE Example</font></a></summary>
+    <br>
+
+**[Legacy] Using Quantum Instance:**
+
+.. code-block:: python
+
+    from qiskit.algorithms import VQE
+    from qiskit.algorithms.optimizers import SLSQP
+    from qiskit.circuit.library import TwoLocal
+    from qiskit.opflow import PauliSumOp
+    from qiskit_aer import AerSimulator
+
+    ansatz = TwoLocal(2, 'ry', 'cz')
+    opt = SLSQP(maxiter=1000)
+
+    # shot-based simulation
+    backend = AerSimulator()
+    qi = QuantumInstance(backend=backend, shots=2048)
+    vqe = VQE(ansatz, optimizer=opt, quantum_instance=qi)
+
+    hamiltonian = PauliSumOp.from_list([("XX", 1), ("XY", 1)]
+    result = vqe.compute_minimum_eigenvalue(hamiltonian)
+
+**[Updated] Using Primitives:**
+
+.. code-block:: python
+
+    # new import!!!
+    from qiskit.algorithms.minimum_eigensolvers import VQE
+    from qiskit.algorithms.optimizers import SLSQP
+    from qiskit.circuit.library import TwoLocal
+    from qiskit.quantum_info import SparsePauliOp
+    from qiskit.primitives import Estimator
+    from qiskit_aer.primitives import Estimator as AerEstimator
+
+    ansatz = TwoLocal(2, 'ry', 'cz')
+    opt = SLSQP(maxiter=1000)
+
+    # shot-based simulation
+    estimator = Estimator(options={"shots": 2048})
+    vqe = VQE(estimator, ansatz, opt)
+
+    # another option
+    aer_estimator = AerEstimator(run_options={"shots": 2048})
+    vqe = VQE(aer_estimator, ansatz, opt)
+
+    hamiltonian = SparsePauliOp.from_list([("XX", 1), ("XY", 1)]
+    result = vqe.compute_minimum_eigenvalue(hamiltonian)
+
+.. raw:: html
+
+   </details>
+   <br>
+
+.. raw:: html
+
+    <details>
+    <summary><a><font size="+1">SamplingVQE Example</font></a></summary>
```

Comment:
Of the title here - would it be better to say something like `VQE applying CVAR` or something - maybe one read the text above and realize what Sampling VQE is already, but maybe it could be more generic description.